### ISSUE TITLE
[apex] New Rule: Repeated Primitive Values

### ISSUE THREAD

**Issue by mitchspano** [pmd/pmd#5323]
**Proposed Rule Name:** RepeatedPrimitiveValues

**Proposed Category:** Code Style

**Description:** This rule aims to identify instances where [primitive data values](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_primitives.htm) (e.g., integers, strings) are repeated multiple times within a method, class, or other scope, without being defined as constants. Such repetition can:

* **Reduce readability:**  Makes it harder to understand the purpose and relationship of the repeated values.
* **Increase maintenance effort:**  If the value needs to be changed, it must be updated in multiple places, increasing the risk of errors.
* **Hinder refactoring:** Makes it difficult to extract logic or modify the code without introducing inconsistencies. 

**Assumptions:**

* This rule considers repetitions of literal primitive values (e.g., `10`, `"hello"`).
* This rule is omitted for  the `Boolean`, `Blob`, and `Object` data types.
* It does not flag certain commonly used  integers (`0`, `1`, `-1`)
* It does not flag any primitive value if it is used only once.

**Limitations:**

* It might not identify all cases of semantically identical values represented in different forms (e.g., `10` and `10.0`).
* It might not identify repetition of strings within concatenation operations (e.g., `'This is a string'` and `'This is a ' + 'string'`).

**Code Sample:** 

**Example:**

```java
public class Example {
  // Violation: The value 5 is repeated multiple times
  public void myMethod() {
    Integer age1 = 5;
    Integer age2 = 5;
    if (someCondition()) {
      age1 = age1 + 5; // Another repetition of 5
    } else {
      age2 = 5; // Yet another repetition of 5
    }
  }
}

// Corrected Code:
public class Example {
  private static final Integer DEFAULT_AGE = 5;
  public void myMethod() {
    Integer age1 = DEFAULT_AGE;
    Integer age2 = DEFAULT_AGE;
    if (someCondition()) {
      age1 = age1 + DEFAULT_AGE;
    } else {
      age2 = DEFAULT_AGE;
    }
  }
}
```

**Possible Properties:**

*   Should this rule be customizable via properties? | No, this is a straightforward rule
